include/thingset.h

```diff
@@ -1227,6 +1249,21 @@ extern "C" {
         _ARRAY_SIZE(((struct_type *)0)->struct_member), \
     };
 
+/**
+ * Define a struct thingset_records to expose a nested array of records.
+ *
+ * @param var_name Name of the created struct thingset_records variable
+ * @param struct_type Type of the struct used for the records (e.g. `struct my_record`)
+ * @param struct_member Struct member used for this item
+ */
+#define THINGSET_DEFINE_RECORD_RECORDS(var_name, struct_type, struct_member) \
+    struct thingset_records var_name = { (void *)offsetof(struct_type, struct_member), \
+                                         sizeof(__typeof__(((struct_type *)0)->struct_member)) \
+                                             / _ARRAY_SIZE(((struct_type *)0)->struct_member), \
+                                         _ARRAY_SIZE(((struct_type *)0)->struct_member), \
+                                         _ARRAY_SIZE(((struct_type *)0)->struct_member), \
```

Comment:
I think we have to, because I don't know how else we'd track their usage. I think once you've got to this level of nesting, the help you get from `thingset_records` tracking used vs. total is probably a bit much to expect.